main.cpp

```diff
@@ -258,12 +259,17 @@ TEST(fileInputClassTest, fullTest){
 
 }
 
-TEST(stdinInputTest, VirtaulGetDataTestSingleLineTest){
+/*TEST(stdinInputTest, VirtaulGetDataTestSingleLineTest){
 
     std::unique_ptr<Input> t1 = std::make_unique<StdinInput>();
     EXPECT_EQ("T est\ne s t\ns t\n\n\n",t1->getData());
-}
+}*/
 
+TEST(TcpIpInputClassTest, virtualGetDataTest){
+    std::unique_ptr<Input> t1 = std::make_unique<TcpIpInput>(8080,AF_INET,INADDR_ANY);
+    t1->getData();
+    EXPECT_EQ(1,1);
+}
```

Comment:
Could you test if `getData()` can be invoked multiple times? I am not 100% sure if it would work.

Reply:
yup, unfortunatelly, it doesn't work in this case. I must do bugfix